pyOCD/target/cortex_m.py

```diff
@@ -942,6 +945,12 @@ def setBreakpoint(self, addr):
         if self.fpb_enabled is False:
             self.enableFPB()
 
+        if addr >= 0x20000000:
+            # Hardware breakpoints are only supported in the range
+            # 0x00000000 - 0x1fffffff on cortex-m devices
+            logging.error('Breakpoint out of range 0x%X', addr)
+            return False
```

Comment:
The COMP field of the FP_COMPn register is only bits 2 to 28 so the max address it supports is 0x20000000.  I found this in the ARMv7-M Architecture Reference Manual and also in the ARMv6-M Architecture Reference Manual.  It never explicitly says breakpoints don't work above address 0x20000000, but there is no way to specify those addresses.  It really surprised me that this was the case, since I had always assumed breakpoints worked everywhere you could run code.  Came across it when testing the flash algo for my last pull request.

Fortunately this doesn't appear to be a problem watchpoints.  I think the DWT supports PC address compare, so it could be uesd to simulate a breakpoint at higher addresses.  It sounds like it is imprecise though, so it might actually halt from the watchpoint a few addresses later.